src/hotspot/share/utilities/globalDefinitions_gcc.hpp

```diff
@@ -48,7 +59,7 @@
 #include <limits.h>
 #include <errno.h>
 
-#if defined(LINUX) || defined(_ALLBSD_SOURCE)
+#if defined(LINUX) || defined(_ALLBSD_SOURCE) || defined(_AIX)
```

Comment:
I cannot answer this question.
If this line is now obsolete it was also obsolete before including AIX, because AIX didn't use this file beforehand.

Reply:
There was at one time an attempt at a gcc/Solaris port, but I think it was
never completed, and most vestiges removed. More recently, @TheShermanTanker
has been doing stuff to permit clang/Windows, and clang-based builds use this file.
I'm kind of surprised he hasn't encountered problems and done some cleanup here.

<inttypes.h> (and <cinttypes>) and 64bit integer types are standard C++ now,
so no longer need all this conditionalization. I suggest cleaning that up as a
separate precursor. That would eliminate the two !defined blocks entirely. I
wish the other conditional includes in this block were "where needed" rather
than in globalDefinitions_gcc.hpp, but that's a different mess.

Reply:
https://bugs.openjdk.org/browse/JDK-8329546 - I can take this if nobody else grabs it soon.

Reply:
I'm waiting for a bunch of tests to complete, so decided to just take that issue.

Reply:
https://github.com/openjdk/jdk/pull/18586

Reply:
@kimbarrett I've been doing things to permit gcc/Windows, not clang. clang has too many different distributions on Windows for me to settle on one, and generalising all of them to be able to compile with any of the Windows clang distributions seamlessly and without issues sounds like a nightmare :P gcc on the other hand has just 2: MSYS2 MINGW64 with ucrt (Which is the one I'm working on) and standalone gcc Windows builds that link to ucrt

I haven't sent a cleanup in this area because I thought my changes were too specific to gcc/Windows, and wouldn't help much with HotSpot in general. I've learnt from my mistakes in the past where I caused reviewers pain in reviewing my admittedly selfish changes to HotSpot :(
That said, if it is requested of me, I can commit some cleanups to this file. What do you think?

Reply:
@TheShermanTanker It depends on the details, of course.  I think there are lots of possible cleanups in this
vicinity that have little or nothing to do with gcc/Windows specifically, though might help there too.

And yeah, I misremembered that it was not clang/Windows but rather gcc/Windows you were working on.
But the same points largely apply here.